website/source/api/discovery-chain.html.md

```diff
@@ -0,0 +1,541 @@
+
+  - `Mode` `(string: "")` - One of `none`, `local`, or `remote`.
+
+### Sample Compilations
+
+#### Multi Datacenter Failover
+
+Config entries defined:
+
+```hcl
+kind            = "service-resolver"
+name            = "web"
+connect_timeout = "15s"
+failover = {
+  "*" = {
+    datacenters = ["dc3", "dc4"]
+  }
+}
+```
+
+Request:
+
+```text
+$ curl http://127.0.0.1:8500/v1/discovery-chain/web
+```
+
+Response:
+
+```json
+{
+    "Chain": {
+        "ServiceName": "web",
+        "Namespace": "default",
+        "Datacenter": "dc1",
+        "Protocol": "tcp",
+        "StartNode": "resolver:web.default.dc1",
+        "Nodes": {
+            "resolver:web.default.dc1": {
+                "Type": "resolver",
+                "Name": "web.default.dc1",
+                "Resolver": {
+                    "ConnectTimeout": 15000000000,
+                    "Target": "web.default.dc1",
+                    "Failover": {
+                        "Targets": [
+                            "web.default.dc3",
+                            "web.default.dc4"
+                        ]
+                    }
+                }
+            }
+        },
+        "Targets": {
+            "web.default.dc1": {
+                "ID": "web.default.dc1",
+                "Service": "web",
+                "Namespace": "default",
+                "Datacenter": "dc1",
+                "MeshGateway": {},
+                "Subset": {}
+            },
+            "web.default.dc3": {
+                "ID": "web.default.dc3",
+                "Service": "web",
+                "Namespace": "default",
+                "Datacenter": "dc3",
+                "MeshGateway": {},
+                "Subset": {}
+            },
+            "web.default.dc4": {
+                "ID": "web.default.dc4",
+                "Service": "web",
+                "Namespace": "default",
+                "Datacenter": "dc4",
+                "MeshGateway": {},
+                "Subset": {}
+            }
+        }
+    }
+}
+```
+
+#### Datacenter Redirect with Overrides
+
+Config entries defined:
+
+```hcl
+kind            = "service-resolver"
+name            = "web"
+redirect {
+  datacenter = "dc2"
+}
+```
+
+Request:
+
+```text
+$ curl -X POST \
+    -d'
+{
+    "OverrideConnectTimeout": "7s",
+    "OverrideProtocol": "grpc",
+    "OverrideMeshGateway": {
+        "Mode": "remote"
+    }
+}
+' http://127.0.0.1:8500/v1/discovery-chain/web
+```
+
+Response:
+
+```json
+{
+    "Chain": {
+        "ServiceName": "web",
+        "Namespace": "default",
+        "Datacenter": "dc1",
+        "CustomizationHash": "b94f529a",
+        "Protocol": "grpc",
+        "StartNode": "resolver:web.default.dc2",
+        "Nodes": {
+            "resolver:web.default.dc2": {
+                "Type": "resolver",
+                "Name": "web.default.dc2",
+                "Resolver": {
+                    "ConnectTimeout": 7000000000,
```

Comment:
is this in nano seconds? I don't think we use that anywhere in our API currently, would it be horrible to convert to milliseconds and possibly name it `ConnectTimeoutMS` or something? Not a big deal but we shoudl maybe at least document the unit here if we leave it.

Reply:
This is a `time.Duration` field. Is there some trickery to get it to serialize better without having to recode the structure?

Reply:
Figured out the trick: https://github.com/hashicorp/consul/pull/6348